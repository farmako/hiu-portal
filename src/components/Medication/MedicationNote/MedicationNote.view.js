import React from 'react';
import * as PropTypes from 'prop-types';

const MedicationNote = (props) => {
  const { mr } = props;
  if (mr.note && mr.note.length > 0) {
    return (
      <li>
        Note:&nbsp;
        <span>
          {mr.note.map((n) => n.text).reduce((acc, value) => `${acc}, ${value}`)}
        </span>
      </li>
    );
  }
  return null;
};

MedicationNote.propTypes = {
  mr: PropTypes.shape({
    note: PropTypes.arrayOf(PropTypes.shape({
      text: PropTypes.string,
    })),
  }),
};

MedicationNote.defaultProps = {
  mr: {
    note: [],
  },
};

export default MedicationNote;
