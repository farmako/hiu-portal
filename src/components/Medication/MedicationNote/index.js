import MedicationNote from './MedicationNote.view';

export default MedicationNote;
