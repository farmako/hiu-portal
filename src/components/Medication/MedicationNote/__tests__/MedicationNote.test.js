import React from 'react';
import {shallow} from "enzyme";
import MedicationNote from "../MedicationNote.view";

describe('MedicationNote', () => {
    const mockData = {
        resourceType: 'MedicationRequest',
        status: 'active',
        note:[
            {
                text:'medication note 1'
            },
            {
                text:'medication note 2'
            }
        ]
    };
    const wrapper = shallow(
      <MedicationNote mr={mockData} />
    );

    it('should render properly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("li").exists()).toEqual(true);
    });

    it('should not render anything if data is undefined', () => {
        wrapper.setProps({mr:undefined});
        expect(wrapper.find("li").exists()).toEqual(false);
    });
});