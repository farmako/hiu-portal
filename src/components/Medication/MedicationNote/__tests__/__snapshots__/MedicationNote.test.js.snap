// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicationNote should render properly 1`] = `
<li>
  Note:Â 
  <span>
    medication note 1, medication note 2
  </span>
</li>
`;
